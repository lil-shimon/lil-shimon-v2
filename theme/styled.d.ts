import "styled-components";

declare module "styled-components" {
  export interface DefaultTheme {
    mediaQuery: {
      sp: string;
      tablet: string;
      laptop: string;
    };
  }
}
