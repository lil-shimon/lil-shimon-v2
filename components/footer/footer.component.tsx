import * as Styles from "./footer.styles";

export const Footer = () => {
  return (
    <Styles.StyledFooter>
      <p>Â© 2024 Shimosawa Kenta. All rights reserved.</p>
    </Styles.StyledFooter>
  );
};
