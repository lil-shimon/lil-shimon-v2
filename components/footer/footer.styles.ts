import styled from "styled-components";

export const StyledFooter = styled.footer`
  display: flex;
  justify-content: center;
  font-size: 0.875rem;
  opacity: 0.4;
`;
