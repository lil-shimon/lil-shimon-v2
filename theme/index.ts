import { DefaultTheme } from "styled-components";

const customMediaQuery = (minWidth: number) =>
  `@media (min-width: ${minWidth}px)`;

export const theme: DefaultTheme = {
  mediaQuery: {
    sp: customMediaQuery(576),
    tablet: customMediaQuery(768),
    laptop: customMediaQuery(992),
  },
};
