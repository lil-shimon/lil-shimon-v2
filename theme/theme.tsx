import { ReactNode } from "react";
import { ThemeProvider as StyledComponentThemeProvider } from "styled-components";

import { DefaultTheme } from "styled-components";

const customMediaQuery = (minWidth: number) =>
  `@media (min-width: ${minWidth}px)`;

const theme: DefaultTheme = {
  mediaQuery: {
    sp: customMediaQuery(576),
    tablet: customMediaQuery(768),
    laptop: customMediaQuery(992),
  },
};

type Props = {
  children: ReactNode;
};

export const ThemeProvider = ({ children }: Props) => {
  return (
    <StyledComponentThemeProvider theme={theme}>
      {children}
    </StyledComponentThemeProvider>
  );
};
