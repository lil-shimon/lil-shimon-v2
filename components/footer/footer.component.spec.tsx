import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import { Footer } from "./footer.component";
import { ThemeProvider } from "styled-components";
import { theme } from "@/theme";

test("should render", () => {
  const { asFragment } = render(<Footer />, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    ),
  });

  expect(asFragment()).toMatchSnapshot();
});
