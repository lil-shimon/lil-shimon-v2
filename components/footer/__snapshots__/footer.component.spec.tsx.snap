// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render 1`] = `
<DocumentFragment>
  <footer
    class="sc-blHHSb cxvegA"
  >
    <p>
      Â© 2024 Shimosawa Kenta. All rights reserved.
    </p>
  </footer>
</DocumentFragment>
`;
